src/RedisResourceManager.php

```diff
@@ -22,17 +22,33 @@
 use function method_exists;
 use function parse_url;
 use function str_replace;
+use function str_starts_with;
 use function strpos;
 use function strtoupper;
 use function trim;
 
 /**
  * This is a resource manager for redis
  */
+/**
+ * @psalm-type ResourceArrayShape =array{
+ *     resource?: RedisResource,
+ *     database?: non-empty-string,
+ *     lib_options?: array,
+ *     version?: int|string,
+ *     password?: non-empty-string,
+ *     persistent_id?: non-empty-string,
+ *     server?: non-empty-string,
+ *     user?: non-empty-string|null,
+ *     initialized?: bool
+ * }
+ */
 final class RedisResourceManager
 {
     /**
-     * Registered resources
+     *  Registered resources
+     *
+     * @var array<ResourceArrayShape> $resources
```

Comment:
```suggestion
     * @var array<string, ResourceArrayShape> $resources
```
the key here should be of the same type of `resource` which seems to be string according to the other methods in this class